stages/org.osbuild.rhsm.facts

```diff
@@ -23,13 +24,21 @@ SCHEMA = r"""
 
 
 def main(tree, options):
+    # `osbuild` itself will write some default facts about its setup into the
+    # facts stage when it is enabled, these are overwritten if supplied by the
+    # options to the stage
+    facts = {
+        "image-builder.osbuild.version": str(osbuild.__version__),
+    }
+    facts.update(options["facts"])
```

Comment:
This changes the behaviour of the manifest without changing the manifest id, breaking caching etc (which matters as the previous version was released to RHEL). Could we just make this opt-in?

That aside, this will produce different images from the same manifest built with different osbuild versions, which is a big no-no. Is there some way to work around this?

Lastly, are we really sure we want to report the osbuild version as an rhsm fact? How and why would this be used? By whom?

Reply:
The alternative to having it always write would be to have a separate property in the manifest that enables writing this (or other osbuild-related) keys, I don't know if that's a particularly good solution (you'd end up with a whole bunch of flags just to change the manifest) but it would solve the same-manifest-different-result.

As for having this fact, I'm going through the list here for making followup PRs: https://github.com/osbuild/osbuild/pull/1060#issuecomment-1189226899, things that were mentioned in the (non-public) google doc, and some discussion on (non-public) slack.

Reply:
Yeah, I agree with Tom: We cannot leak ambient information into the image, since we clearly specify that the manifest is the sole source of information of what ends up in the image.

A simple way out is to make the caller include the osbuild version as value in the options to this stage. So the value does not come from osbuild, but from the caller. You could technically annotate it in a way for osbuild to verify the value matches its version, but I think this would not be needed.

But I really wonder why we want that information anyway? Is this just a "nice to have for debugging"?